DOC-5902 - default revs_limit value

jamesnocentini opened the pull request:
https://issues.couchbase.com/browse/DOC-5902

jamesnocentini committed:
DOC-5902 - default revs_limit value

https://issues.couchbase.com/browse/DOC-5902

bbrks commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -521,7 +521,7 @@ properties:
               - Sync Gateway purge endpoint [/{db}/_purge](admin-rest-api.html#/document/post__db___purge).
               - Sync Gateway [document TTLs](admin-rest-api.html#/document/put__db___doc_).
-            default: 1000
+            default: 50
```
50 when allow_conflicts=false, 100 otherwise. Maybe just include that info in the description instead, and keep this value as 50.

Also include allow conflicts disclaimer in description:
```diff
Maximum depth to which a document's revision tree can grow.
In Sync Gateway 1.5, the `revs_limit` configuration cannot be set lower than 20.
-Setting the `revs_limit` to a low value (typically below 100) can have adverse effects as described below.
+Setting the `revs_limit` to a low value (typically below 100) when allow_conflicts is enabled can have adverse effects as described below.
```

jamesnocentini replied:
- Callout that the default value is lowered to 50 if allow_conflicts is disabled
- Callout the warning for a value lower than 100 as only applicable if allow_conflicts is enabled

https://github.com/couchbase/docs-sync-gateway/commit/47577088cfad91777a42b600870f4ecacf088ad2

OK, I've kept the default value to 1000 (default with `allow_conflicts`'s default). I'll re-request a review.

bbrks replied:
Default in 2.6 with allow_conflicts=true is now down to 100 (one hundred)

jamesnocentini replied:
- Callout behavior change in release notes
- Edit first sentences
- Add title for advanced topic about disconnected branches

https://github.com/couchbase/docs-sync-gateway/commit/16184cd754d5bc1b05f73919dc648b06d926ab6e

Thanks, I've re-worked it slightly. I'll branch out non-2.6 edits in a separate PR.

jamesnocentini replied:
Should be good to merge now, unless you spot anything else.

jamesnocentini committed:
- Callout that the default value is lowered to 50 if allow_conflicts is disabled
- Callout the warning for a value lower than 100 as only applicable if allow_conflicts is enabled

jamesnocentini committed:
- Callout behavior change in release notes
- Edit first sentences
- Add title for advanced topic about disconnected branches

jamesnocentini committed:
- Non-2.6 changes moved to https://github.com/couchbase/docs-sync-gateway/pull/125
- Revert here

bbrks commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -506,7 +506,7 @@ properties:
             type: integer
             description: |
-              Maximum depth to which a document's revision tree can grow. In Sync Gateway 1.5, the `revs_limit` configuration cannot be set lower than 20. Setting the `revs_limit` to a low value (typically below 100) can have adverse effects as described below.
+              Maximum depth to which a document's revision tree can grow. The default value for `revs_limit` is 100 if [allow_conflict](config-properties.html#databases-foo_db-allow_conflicts) is enabled (default) and 50 if `allow_conflict` is disabled. The minimum value is 20. Setting the `revs_limit` to a low value (typically below 100) can have adverse effects as described below.
```
```suggestion
              Maximum depth to which a document's revision tree can grow. The default value for `revs_limit` is 100 if [allow_conflicts](config-properties.html#databases-foo_db-allow_conflicts) is enabled (default) and 50 if `allow_conflicts` is disabled. The minimum value is 20. Setting the `revs_limit` to a low value (typically below 100) can have adverse effects as described below.
```

jamesnocentini replied:
👍

bbrks reviewed (dismissed):
Looks alright to me after pluralising the `allow_conflicts` config option as suggested

Commit:
Update modules/ROOT/assets/attachments/sg.yaml

Co-Authored-By: Ben Brooks <ben.brooks@couchbase.com>

ibsoln commented:
@rajagp, @bbrks , @sarathkumarsivan, @adamcfraser - can this now be approved so it can be merged?
